src/ui/views/strategies/operations/document.rs

```diff
@@ -30,22 +28,39 @@ pub(super) struct StrategyOpDocumentFormController {
         Field<SelectInput<u16>>,
         Field<SelectInput<f64>>,
     )>,
-    selected_strategy: String,
+    selected_strategy_name: String,
     known_contracts: BTreeMap<String, DataContract>,
     supporting_contracts: BTreeMap<String, DataContract>,
     document_types: BTreeMap<String, DocumentType>,
+    strategy_contract_names: StrategyContractNames,
 }
 
 impl StrategyOpDocumentFormController {
     pub(super) fn new(
-        selected_strategy: String,
+        selected_strategy_name: String,
         known_contracts: BTreeMap<String, DataContract>,
         supporting_contracts: BTreeMap<String, DataContract>,
+        strategy_contract_names: StrategyContractNames,
     ) -> Self {
         // Collect known_contracts and supporting_contracts names for the form
         let mut contract_names: Vec<String> = known_contracts.keys().cloned().collect();
-        contract_names.extend(supporting_contracts.keys().cloned());
 
+        // Add only supporting contracts that are also in strategy_contract_names
+        for (supporting_contract_name, _) in supporting_contracts.iter() {
+            if strategy_contract_names.iter().any(|(name, _)| name == supporting_contract_name) {
```

Comment:
but in general I think we better iterate `strategy_contract_names` vector here because it's sequential and unordered and for each item check for existence in `supporting_contracts` because it will do binary search since keys are ordered in btreemap